internal/terraform/module/module_ops_test.go

```diff
@@ -1088,6 +1089,85 @@ func TestParseModuleConfiguration_ignore_tfvars(t *testing.T) {
 	}
 }
 
+func TestParseVariables(t *testing.T) {
+	ctx := context.Background()
+	ss, err := state.NewStateStore()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testData, err := filepath.Abs("testdata")
+	if err != nil {
+		t.Fatal(err)
+	}
+	testFs := filesystem.NewFilesystem(ss.DocumentStore)
+
+	singleFileModulePath := filepath.Join(testData, "single-file-change-module")
+
+	err = ss.Modules.Add(singleFileModulePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ctx = lsctx.WithLanguageId(ctx, ilsp.Tfvars.String())
+	ctx = lsctx.WithRPCContext(ctx, lsctx.RPCContextData{})
+	err = ParseModuleConfiguration(ctx, testFs, ss.Modules, singleFileModulePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = ParseVariables(ctx, testFs, ss.Modules, singleFileModulePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	before, err := ss.Modules.ModuleByPath(singleFileModulePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// say we're coming from did_change request
+	fooURI, err := filepath.Abs(filepath.Join(singleFileModulePath, "example.tfvars"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	x := lsctx.RPCContextData{
+		Method: "textDocument/didChange",
+		URI:    uri.FromPath(fooURI),
+		// URI:    "file:///test/example.tfvars",
+	}
+
+	// ignore job state
+	ctx = job.WithIgnoreState(ctx, true)
+	ctx = lsctx.WithLanguageId(ctx, ilsp.Tfvars.String())
+	ctx = lsctx.WithRPCContext(ctx, x)
+	err = ParseVariables(ctx, testFs, ss.Modules, singleFileModulePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	after, err := ss.Modules.ModuleByPath(singleFileModulePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// example.tfvars should not be the same as first seen
+	if before.ParsedVarsFiles["example.tfvars"] == after.ParsedVarsFiles["example.tfvars"] {
+		t.Fatal("file should mismatch")
+	}
+
+	b := before.VarsDiagnostics[ast.HCLParsingSource]
+	ex := b[ast.VarsFilename("example.tfvars")]
+
+	a := after.VarsDiagnostics[ast.HCLParsingSource]
+	exA := a[ast.VarsFilename("example.tfvars")]
+
+	// // diags should change for example.tfvars
+	if ex[0] == exA[0] {
+		// if before.VarsDiagnostics["example.tfvars"][0] == after.VarsDiagnostics["example.tfvars"][0] {
+		t.Fatal("diags should mismatch")
+	}
```

Comment:
I think this test would pass even prior to the implementation change. The main difference we should be testing here, in addition to what's being tested, is whether other (`*.tfvars`) file which was _not_ opened has _not_ changed.

Would you mind adding that to the test? It also implies adding another `*.tfvars` file, possibly with diagnostics as well, which can also be compared and should match.